solr/core/src/java/org/apache/solr/schema/NVectorField.java

```diff
@@ -0,0 +1,285 @@
+import org.apache.solr.response.TextResponseWriter;
+import org.apache.solr.search.QParser;
+import org.apache.solr.uninverting.UninvertingReader;
+import org.apache.solr.util.NVectorUtil;
+
+import java.io.IOException;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+public class NVectorField extends CoordinateFieldType {
+
+    String DEFAULT_SEPARATOR = ",";
+    String separator = DEFAULT_SEPARATOR;
+
+    @Override
+    protected void init(IndexSchema schema, Map<String, String> args) {
+        super.init(schema, args);
+        separator = args.getOrDefault("separator", DEFAULT_SEPARATOR);
+        dimension = 3;
+        createSuffixCache(3);
+    }
+
+    @Override
+    public List<IndexableField> createFields(SchemaField field, Object value) {
+        String externalVal = value.toString();
+        String[] point = parseCommaSeparatedList(externalVal, dimension, separator);
+        String[] nvector;
+        try {
+            NumberFormat format = NumberFormat.getInstance(Locale.getDefault());
+            format.setParseIntegerOnly(false);
+            nvector = NVectorUtil.latLongToNVector(point, format);
+        } catch (ParseException e) {
+            throw new SolrException(
+                    SolrException.ErrorCode.BAD_REQUEST,
+                    "format exception parsing: "+externalVal);
+        }
+        List<IndexableField> f = new ArrayList<>((dimension * 2) + 1);
+
+        if (field.indexed()) {
+            for (int i = 0; i < dimension; i++) {
+                SchemaField sf = subField(field, i, schema);
+                f.addAll(sf.createFields(nvector[i]));
+            }
+        }
+
+        if (field.stored()) {
+            f.add(createField(field.getName(), externalVal, StoredField.TYPE));
+        }
+        return f;
+    }
+
+    @Override
+    public ValueSource getValueSource(SchemaField field, QParser parser) {
+        ArrayList<ValueSource> vs = new ArrayList<>(dimension);
+        for (int i = 0; i < dimension; i++) {
+            SchemaField sub = subField(field, i, schema);
+            vs.add(sub.getType().getValueSource(sub, parser));
+        }
+        return new NVectorValueSource(vs);
+    }
+
+    /**
+     * Given a string containing <i>dimension</i> values encoded in it, separated by commas, return a
+     * String array of length <i>dimension</i> containing the values.
+     *
+     * @param externalVal The value to parse
+     * @param dimension   The expected number of values for the point
+     * @param separator   The separator between values
+     * @return An array of the values that make up the point (aka vector)
+     * @throws SolrException if the dimension specified does not match the number found
+     */
+    public static String[] parseCommaSeparatedList(String externalVal, int dimension, String separator)
```

Comment:
Can we have some unit tests for this? Parsing is one of those thing that seems easy but there are always edge cases and it's easy to introduce regressions.